frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -894,3 +895,175 @@ export function ConditionsSection(props: { resource: KubeObjectInterface | null
+        </Link>
+      );
+    } else if (configMap) {
+      return (
+        <Link routeName="configMap" params={{ namespace: namespace, name: configMap.name }}>
+          {configMap.name}
+        </Link>
+      );
+    } else {
+      return volumeName;
+    }
+  }
+
+  function volumeRows(volume: VolumeSectionProps['volume']) {
+    const currentVolume = volume;
+    if (!currentVolume) {
+      return [];
+    }
+
+    let volumeName = currentVolume.name;
+    let volumeType: string = '';
+
+    const volumeKeys = Object.keys(currentVolume);
+
+    for (const key of volumeKeys) {
+      if (key === 'name') {
+        volumeName = currentVolume[key];
+      } else {
+        volumeType = key;
+      }
+    }
+
+    function volumeDetailGrab(vol: VolumeSectionProps['volume']) {
+      // array for items that are printable
+      const directPrint = [];
+
+      // array for items that are not printable and need to be printed to yaml
+      const yamlPrint = [];
+
+      // loop over volumeKeys and check if the value is a string, number, or bool
+      for (const key in vol) {
+        if (!(vol[key] === '')) {
+          if (
+            typeof vol[key] === 'string' ||
+            typeof vol[key] === 'number' ||
+            typeof vol[key] === 'boolean'
+          ) {
+            directPrint.push({
+              volKey: key,
+              volValue: vol[key],
+            });
+          } else {
+            yamlPrint.push({
+              volKey: key,
+              volValue: vol[key],
+            });
+          }
+        }
+      }
+
+      const volumePrints = {
+        directPrint,
+        yamlPrint,
+      };
+
+      return volumePrints;
+    }
+
+    const volumeDetails: any = volumeDetailGrab(currentVolume[volumeType]);
+
+    function yamlPrinter(yamlData: any) {
+      return (
+        <Typography component="pre" variant="body2">
+          {YAML.stringify(yamlData)}
+        </Typography>
+      );
+    }
+
+    return [
+      {
+        name: volumeName,
+        withHighlightStyle: true,
+      },
+      ...(volumeType
+        ? [
+            {
+              name: 'Kind',
+              value: volumeType,
+            },
+          ]
+        : []),
+      {
+        name: 'Source',
+        value: printVolumeLink(volumeName, currentVolume),
+      },
+      ...(volumeDetails.directPrint
+        ? [
+            ...volumeDetails.directPrint.map((detail: any) => {
+              return {
+                name: detail.volKey,
+                value: detail.volValue,
+              };
+            }),
+          ]
+        : []),
+      ...(volumeDetails.yamlPrint
+        ? [
+            ...volumeDetails.yamlPrint.map((detail: any) => {
+              return {
+                name: detail.volKey,
+                value: yamlPrinter(detail.volValue),
+              };
+            }),
+          ]
+        : []),
+    ];
+  }
+
+  return (
+    <SectionBox title={t('frequent|Volumes')}>
```

Comment:
When you rebase this PR. Please remove the namespace unless this word belongs to the glossary namespace.